libyul/backends/wasm/polyfill/Keccak.yul

```diff
@@ -18,7 +18,332 @@
 
 // NOTE: This file is used to generate `ewasmPolyfills/Keccak.h`.
 
+// reset the context
+function keccak_reset(context_offset:i32) {
+	// state context
+	//	typedef struct {
+	//		union {                                 // state:
+	//			uint8_t b[200];                     // 8-bit bytes
+	//			uint64_t q[25];                     // 64-bit words
+	//		} st;
+	//		int pt, rsiz, mdlen;                    // these don't overflow
+	//	} sha3_ctx_t;
```

Comment:
These code snippets in comments come from https://github.com/mjosaarinen/tiny_sha3, right? I think it would be best to have an acknowledgement and a link to it in a comment at the top of the file.